test/functional/p2p_addr_relay.py

```diff
@@ -101,6 +115,25 @@ def relay_tests(self):
 
         self.nodes[0].disconnect_p2ps()
 
+    def blackhole_tests(self):
```

Comment:
nice tests, would it would be overkill to also test for the outbound case?

Reply:
definitely not overkill, the inbound vs outbound behavior is different- inbounds we wait to hear an `addr` message before considering the peer eligible for relay, but with outbounds we setup relay, send `getaddr` & self-announce (unless its a block-relay-only connection). I added some tests to check these behaviors.